src/App.jsx

```diff
@@ -49,7 +50,43 @@ export function App() {
 			setTimeOutErrorMsg('New List Error. Please refresh or try again later.');
 		}
 	};
-
+	/**
+	 * Callback function gets passed as a prop through Home component to retrieve user input.
+	 * A Promise.race() method is used again as with setList to appropriately respond to the user if the database is not responding.
+	 * StreamListItems function takes the user submitted token and returns a snapshot of the items in the database.
+	 * If the collection is not empty, the shared token is set to localStorage.
+	 * Else the user is informed the token that was shared with them is empty or does not exist via setJoinListErrorMsg state that is passed through Home component.
+	 * **/
+	const joinList = async (token) => {
+		try {
+			await Promise.race([
+				streamListItems(token, (snapshot) => {
```

Comment:
Instead of creating a new function for a get request, I used the existing streamListItems function to query Firestore for the collection and filter for lists that were not empty.

Reply:
Is there a plan for when you find yourself repeating this .race condition?